### PR TITLE
feat: Set up a database and contact persistence.

### PR DESCRIPTION
**Pull request by robinlinden**
Not sure if we want to check in the generated code or not. It seems to be 50/50 what people do in the Flutter dev community.

I currently have this PR set up with all generated files ignored, with scripts to generate them under tools/. We'll need some form of preparation scripts there even if we check in the generated files as different builds need different things downloaded. E.g. the web build needs the wasm/js files from 3rd-party libraries, and Windows needs sqlite3.dll.

See:
https://stackoverflow.com/questions/56110386/should-i-commit-generated-code-in-flutter-dart-to-vcs
https://pub.dev/packages/build_runner#source-control

This change is [Reviewable](https://reviewable.io/reviews/toktok/btox/20)

### CODE REVIEW

**Comment by robinlinden** [comment]
You can view the database used on the preview page in the Firefox dev tools here:
drift database location

File: lib/contact_list_page.dart [hunk 1]
```diff
@@ -4,10 +4,10 @@ import 'package:flutter/services.dart';
 
 import 'add_contact_page.dart';
 import 'chat_page.dart';
-import 'contact.dart';
 import 'profile.dart';
 import 'settings.dart';
 import 'strings.dart';
+import 'db/database.dart';
```
Comment (iphydf) [inline lib/contact_list_page.dart:10]: Is there some convention of sorting "/" paths after others?

Comment (robinlinden) [reply lib/contact_list_page.dart:10]: No idea. I'll alphabetize them for now and we'll just go with whatever https://dart-lang.github.io/linter/lints/directives_ordering.html wants when we enable it. It sounds like they should just be alphabetized, but automatic enforcement is good.

**Review by JFreegman — APPROVED** [review]
> Reviewed 21 of 21 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/toktok/btox/20)* status: 0 of 1 approvals obtained (waiting on @iphydf)

**Review by iphydf — APPROVED** [review]

**Pull request merged**